dangerzone/isolation_provider/qubes.py

```diff
@@ -84,50 +84,37 @@ def _convert(
         percentage = 0.0
 
         with open(document.input_filename, "rb") as f:
-            # TODO handle lack of memory to start qube
-            if getattr(sys, "dangerzone_dev", False):
-                # Use dz.ConvertDev RPC call instead, if we are in development mode.
-                # Basically, the change is that we also transfer the necessary Python
-                # code as a zipfile, before sending the doc that the user requested.
-                p = subprocess.Popen(
-                    ["/usr/bin/qrexec-client-vm", "@dispvm:dz-dvm", "dz.ConvertDev"],
-                    stdin=subprocess.PIPE,
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE,
-                )
+            p = self.qrexec_subprocess()
+            try:
                 assert p.stdin is not None
-
-                # Send the dangerzone module first.
-                self.teleport_dz_module(p.stdin)
-
-                # Finally, send the document, as in the normal case.
                 p.stdin.write(f.read())
                 p.stdin.close()
-            else:
-                p = subprocess.Popen(
-                    ["/usr/bin/qrexec-client-vm", "@dispvm:dz-dvm", "dz.Convert"],
-                    stdin=f,
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.DEVNULL,
-                )
+            except BrokenPipeError as e:
+                # Likely that qube did not have enough RAM to start
+                # in which case, it exits with code 126
+                if running_on_qubes() and p.wait() == 126:
```

Comment:
Nitpick, but if we are here, should we even check `running_on_qubes()`? Also, if the process has closed its stdin, but has not finished, then execution will hang in this line. I suggest setting a timeout of 3 seconds here.

Reply:
Duh. Yeah, obviously we don't need to. Somehow I muse have imagined this was shared code.

> I suggest setting a timeout of 3 seconds here.

You're right. I also wanted that. I recall me wanting to reuse you timeout code but not doing it because something was still missing or we need to adapt something to make use of that here. I'll recheck.